doc/stdenv/platform-notes.chapter.md

```diff
@@ -46,22 +46,195 @@ Some common issues when packaging software for Darwin:
   }
   ```
 
-- Some packages assume xcode is available and use `xcrun` to resolve build tools like `clang`, etc. This causes errors like `xcode-select: error: no developer tools were found at '/Applications/Xcode.app'` while the build doesn’t actually depend on xcode.
+- Some packages assume Xcode is available and use `xcrun` to resolve build tools like `clang`, etc. The Darwin stdenv includes `xcrun`, and it will return the path to any binary available in a build.
 
   ```nix
   stdenv.mkDerivation {
     name = "libfoo-1.2.3";
     # ...
+    nativeBuildInputs = [ bison ];
+    buildCommand = ''
+      xcrun bison foo.y # produces foo.tab.c
+      # ...
+    '';
+  }
+  ```
+  The package `xcbuild` can be used to build projects that really depend on Xcode. However, this replacement is not 100% compatible with Xcode and can occasionally cause issues.
+
+  Note: Some packages may hardcode an absolute path to `xcrun`, `xcodebuild`, or `xcode-select`. Those paths should be removed or replaced.
+
+  ```nix
+  stdenv.mkDerivation {
+    name = "libfoo-1.2.3";
     prePatch = ''
       substituteInPlace Makefile \
-          --replace-fail '/usr/bin/xcrun clang' clang
+        --replace-fail /usr/bin/xcrun xcrun
+        # or: --replace-fail /usr/bin/xcrun '${lib.getExe' buildPackages.xcbuild "xcrun"}'
     '';
   }
   ```
 
-  The package `xcbuild` can be used to build projects that really depend on Xcode. However, this replacement is not 100% compatible with Xcode and can occasionally cause issues.
+- When a package indicates a minimum supported version, also called the deployment target, you can set it in your derivation using `darwinMinVersionHook`.
```

Comment:
I think that we should mention this after the SDK versions, because anything that needs the deployment target bumping will presumably need the SDK bumping too. Overriding the SDK is more common than the deployment target, so it should be earlier. It may also be easier to lead in to a discussion of SDK version vs. deployment target if we start with the concept of multiple SDK versions.